File: Cap 05/5.3 - Laco Condicional Pre-Teste/5.3.1 - Controle Condicional Verdadeiro/Efetivo/Interativo/enquantoINT.vb
' Livro .....: Algorimos
'              Logica para Desenvolvimento de Programacao de 
'              Computadores
' Autores ...: Jose Augusto N. G. Manzano
'              Jayr Figueiredo de Oliveira
' Editora ...: Erica
' 
' Capitulo ..: 5 - Topico 5.3.1
' 
' Programa ..: LACO CONDICIONAL PRE-TESTE COM FLUXO VERDADEIRO INTERATIVO

Module LoopPreTest

    Sub Main()

        Dim N As Integer
        Dim R As Integer
        Dim RESP As String
        Console.WriteLine("LACO PRE-TESTE COM FLUXO VERDADEIRO")
        Console.WriteLine("Multiplicacao de numero por 3 por N vezes")
        Console.WriteLine()
        RESP = "S"
        Do While (RESP = "S")
            Console.Write("Entre valor numerico inteiro: ")
            N = Console.ReadLine()
            R = N * 3
            Console.WriteLine("O numero informado X 3 = " & R)
            Console.WriteLine()
            Console.Write("Deseja continuar? (S/N): ")
            RESP = Console.ReadLine().ToUpper()
            Console.WriteLine()
        Loop

    End Sub

End Module

File: Cap 05/5.3 - Laco Condicional Pre-Teste/5.3.1 - Controle Condicional Verdadeiro/Simulado/Interativo/enquantoINT.go
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 5 - Topico 5.3.1
 * 
 * Programa ..: LACO CONDICIONAL PRE-TESTE COM FLUXO VERDADEIRO INTERATIVO
 */

package main

import (
  "fmt"
  "strings"
)

var (
  N    int
  R    int
  RESP string
)
func main() {
  fmt.Println("LACO PRE-TESTE COM FLUXO VERDADEIRO")
  fmt.Println("Multiplicacao de numero por 3 por N vezes")
  fmt.Println()
  RESP = "S"
  for (strings.ToUpper(RESP) == "S") { // simulacao "enquanto (RESP = "S") faca"
    fmt.Print("Entre valor numerico inteiro: ")
    fmt.Scanln(&N)
    R = N * 3
    fmt.Println("O numero informado X 3 =", R)
    fmt.Println()
    fmt.Print("Deseja continuar? (S/N): ")
    fmt.Scanln(&RESP)
    fmt.Println()
  } // simulacao "fim_enquanto"
}
